logstash-core/src/main/java/org/logstash/instrument/metrics/Metric.java

```diff
@@ -88,4 +59,16 @@ default String type() {
         return getType().asString();
     }
 
+    /**
+     * Determine if this metric has a value explicitly set.
+     * @return true if this metric has been set to a specific value, false if it is the default state
+     */
+    boolean isDirty();
```

Comment:
I am trying to match the serialized output of the Ruby code exactly _or as close as possible_.

Since the Ruby metrics are lazily added, when serialized they only show up if they have been set. Since the Java metrics are defaulted to null or zero's, those null's or zero's appear in the JSON output. 

By checking the dirty flag during serialization,  I can emulate the lazily loaded metrics by only serializes those metrics which are dirty.